xenonpy/inverse/base.py

```diff
@@ -192,35 +211,58 @@ def log_likelihood(self, X, **kwargs):
         results = []
 
         X, _ = self._check_input(X, **kwargs)
-        for k, lls in self.__loglikelihood_sets__.items():
-            if k in kwargs:
-                k = kwargs[k]  # what is this for? even if k not in kwargs... nothing happen right?
-            if k in X:
+        if isinstance(X, list):
+            for k, lls in self.__loglikelihood_sets__.items():
+                # if k in kwargs:
+                #     k = kwargs[k]  # what is this for? even if k not in kwargs... nothing happen right?
```

Comment:
this line temporarily replaces the col name in the previous version. In your version, it has no means. 

for example:
```py
self.g1 = ....
self.g1 = ...
```
The group `g1` will try to find the col name **g1** in the input dataframe. When you call `.transform(..., g1=other_key)`, then just in this call, group `g1` will try to find col `other_key`

Reply:
I still do not quite understand, maybe you can explain to me in person next time.